tests/ha/haproxy.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test haproxy resource agent
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils 'zypper_call';
+use utils 'systemctl';
+use hacluster;
+
+sub run {
+    my $cluster_name = get_cluster_name;
+    my $haproxy_rsc  = 'haproxy';
+    my $haproxy_cfg  = '/etc/haproxy/haproxy.cfg';
+    my $apache_file  = '/srv/www/htdocs/index.html';
+    my $vip_ip       = '10.0.2.20';
+    my $vip_rsc      = 'vip';
+    my $node_01      = choose_node(1);
+    my $node_02      = choose_node(2);
+    my $node_01_ip   = script_output "host -t A $node_01 | awk '{ print \$NF }'";
+    my $node_02_ip   = script_output "host -t A $node_02 | awk '{ print \$NF }'";
```

Comment:
Per `script_output` documentation, its result could include extra serial output which was on the serial console, so I don't think a simple assignment will cover all bases; I think it would be better to assign the full output into an auxiliary variable, and then use regexps on that variable to get the IP addresses and store them in `$node_01_ip` and `$node_02_ip`.